Client/WebAuthN/U2FExtensions.swift

```diff
@@ -758,70 +919,93 @@ class U2FExtensions: NSObject {
+
         guard let registerRequest = YKFKeyU2FRegisterRequest(challenge: request.challenge, appId: request.appId ?? "") else {
             sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.bad_request)
             return
         }
-        
-        guard let u2fservice = YubiKitManager.shared.keySession.u2fService else {
-            sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error)
-            return
-        }
-        
-        u2fservice.execute(registerRequest) { [weak self] response, error in
+        executeKeyRequestWith { [weak self] in
             guard let self = self else {
-                log.error(U2FErrorMessages.Error.rawValue)
                 return
             }
             
-            guard error == nil else {
-                let errorMessage = error?.localizedDescription ?? Strings.U2FRegistrationError
-                self.sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error, errorMessage: errorMessage)
-                return
-            }
-            guard let clientData = response?.clientData.websafeBase64String() else {
-                self.sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error)
-                return
-            }
-            guard let registrationData = response?.registrationData.websafeBase64String() else {
+            guard let u2fservice = self.keyU2fService else {
                 self.sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error)
                 return
             }
             
-            self.cleanupFIDORegistration(handle: handle)
-            if requestId >= 0 {
-                ensureMainThread {
-                    self.tab?.webView?.evaluateJavaScript("u2f.postLowLevelRegister(\(requestId), \(true), '\(request.version)', '\(registrationData)', '\(clientData)', \(defaultErrorCode), '')", completionHandler: { _, error in
-                        if error != nil {
-                            let errorDescription = error?.localizedDescription ?? U2FErrorMessages.ErrorRegistration.rawValue
-                            log.error(errorDescription)
+            u2fservice.execute(registerRequest) { [weak self] response, error in
+                guard let self = self else {
+                    log.error(U2FErrorMessages.Error.rawValue)
+                    return
+                }
+                
+                guard error == nil else {
+                    let errorMessage = error?.localizedDescription ?? Strings.U2FRegistrationError
+                    self.sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error, errorMessage: errorMessage)
+                    return
+                }
+                guard let clientData = response?.clientData.websafeBase64String() else {
```

Comment:
`clientData` and `registrationData` could go under the same `guard` condition, both return the same error